waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/AbstractThriftMetastoreClientManager.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (C) 2016-2024 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.client;
+
+import java.io.Closeable;
+import java.net.URI;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.conf.HiveConf.ConfVars;
+import org.apache.hadoop.hive.conf.HiveConfUtil;
+import org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore;
+import org.apache.thrift.TException;
+import org.apache.thrift.transport.TTransport;
+
+import lombok.extern.log4j.Log4j2;
+
+import com.hotels.bdp.waggledance.client.compatibility.HiveCompatibleThriftHiveMetastoreIfaceFactory;
+
+@Log4j2
+public abstract class AbstractThriftMetastoreClientManager implements Closeable {
+
+
+  static final AtomicInteger CONN_COUNT = new AtomicInteger(0);
```

Comment:
nit: it would be better to use the full word, `CONNECTION_COUNT`.

Reply:
These things are mostly how it was before it was an abstract class: https://github.com/ExpediaGroup/waggle-dance/pull/313/files#diff-7d969aea494084a6e71ce35f92df70de7b69337a9e08ef955d3ac2b5c4dc9170
Let's leave it, will me easier to merge master branch in at some point.